### PR TITLE
Refactor NFC availability handling

### PR DESCRIPTION
**Pull request by FabioTacke**

### CODE REVIEW

File: app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt [hunk 1]
```diff
@@ -143,15 +129,19 @@ private fun PreviewNfc() {
 @Preview(name = "Large", showSystemUi = true, device = Devices.PIXEL_4_XL)
 @Composable
 private fun PreviewNoNfc() {
-    UseIDApp(appCoordinator = PreviewAppCoordinator(rememberUpdatedState(newValue = NfcAvailability.NoNfc), false), PreviewAppNavigator(), PreviewTrackerManager())
+    UseIDApp(
+        NfcAvailability.Available,
```
Comment (UrsKahmann) [inline app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt:133]: Shouldn't this be `.NoNfc` ?

File: app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt [hunk 2]
```diff
@@ -143,15 +129,19 @@ private fun PreviewNfc() {
...
+    )
 }
 
 @Preview(name = "Small", showSystemUi = true, device = Devices.NEXUS_5)
 @Preview(name = "Large", showSystemUi = true, device = Devices.PIXEL_4_XL)
 @Composable
 private fun PreviewNfcDeactivated() {
     UseIDApp(
-        appCoordinator = PreviewAppCoordinator(rememberUpdatedState(newValue = NfcAvailability.Deactivated), false),
+        NfcAvailability.Available,
```
Comment (UrsKahmann) [inline app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt:144]: Shouldn't this be `.Deactivated` ?

File: app/src/main/kotlin/de/digitalService/useID/ui/coordinators/AppCoordinator.kt [hunk 3]
```diff
@@ -54,26 +64,34 @@ class AppCoordinator @Inject constructor(
...
+
+    private fun handleTcTokenUrl(url: String) {
+        navigator.popToRoot()
+
+        if (nfcInterfaceManager.nfcAvailability.value != NfcAvailability.Available) {
+            logger.debug("Blocking identification due to unavailable NFC.")
+            cachedTcTokenUrl = url
+            return
+        } else {
+            cachedTcTokenUrl = null
```
Comment (UrsKahmann) [inline app/src/main/kotlin/de/digitalService/useID/ui/coordinators/AppCoordinator.kt:85]: I am not sure if I misunderstand this, but shouldn't the two lines be switched? If Nfc is not available -> set the cachedTcTokenUrl to null?

**Review by UrsKahmann — COMMENTED** [review]
Very nice!

File: app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt (hunk 1 above)
Comment (FabioTacke) [reply app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt:133]: No, in the Preview we want to simulate any NFC related interaction. The app thinks NFC is available (and therefore doesn't show the NoNFC screen).

File: app/src/main/kotlin/de/digitalService/useID/ui/coordinators/AppCoordinator.kt (hunk 3 above)
Comment (FabioTacke) [reply app/src/main/kotlin/de/digitalService/useID/ui/coordinators/AppCoordinator.kt:85]: If NFC is not available we want to cache the URL for later (after it has been enabled again eventually). If it is then enabled later the cached URL should be reset to null.

File: app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt (hunk 2 above)
Comment (FabioTacke) [reply app/src/main/kotlin/de/digitalService/useID/ui/UseIDApp.kt:144]: Absolutely! Thanks!

**Review by UrsKahmann — APPROVED** [review]

**Pull request merged**